File: hacking_with_ruby/cms.rb
require 'open-uri'
require 'nokogiri'
require 'colorize'

def test_vulnerability(url)
  puts "Testing CMS Information Disclosure Vulnerabilities for #{url}...".yellow
  begin
    html = URI.open(url).read
    doc = Nokogiri::HTML(html)
    meta_tags = doc.css('meta[name]').map { |meta| meta['name'] }
    if meta_tags.include?('generator')
      generator_content = doc.at('meta[name="generator"]')['content']
      puts "Vulnerability Found!".red
      puts "The website is disclosing CMS information:".red
      puts "Generator meta tag found with content: #{generator_content}".red
      puts "To exploit this vulnerability, an attacker can perform fingerprinting to determine the CMS version and potentially exploit known vulnerabilities.".red
    else
      puts "No vulnerability found.".green
    end
  rescue StandardError => e
    puts "An error occurred while testing for vulnerability: #{e.message}".red
  end
end

print "Enter your target website url: "
target_url = gets.chomp

test_vulnerability(target_url)

File: hacking_with_python/cms.py
import requests
from bs4 import BeautifulSoup
import re
import sys

# Function to check for CMS Information Disclosure Vulnerabilities
def check_cms_info_disclosure(target_url):
    try:
        print("\n[*] Testing for CMS Information Disclosure Vulnerabilities on:", target_url)
        
        # Fetching the website content
        response = requests.get(target_url)
        
        if response.status_code == 200:
            html_content = response.text
            
            # Parsing the HTML content
            soup = BeautifulSoup(html_content, 'html.parser')
            
            # Finding meta generator tag
            meta_generator = soup.find('meta', {'name': re.compile(r'generator', re.I)})
            
            if meta_generator:
                print("\n[*] Found meta generator tag:")
                print(meta_generator)
                print("\n[!] The website may be disclosing CMS information.")
                print("\n[*] To test this vulnerability, try accessing the following URLs:")
                print("\n[1] /wp-admin")
                print("[2] /administrator")
                print("[3] /admin")
                print("[4] /wp-login.php")
                print("[5] /wp-admin/install.php")
            else:
                print("\n[*] No meta generator tag found. The website does not seem to disclose CMS information.")
        else:
            print("\n[!] Failed to retrieve website content. Status Code:", response.status_code)
    except Exception as e:
        print("\n[!] An error occurred:", str(e))

# Function to run the program
def run():
    print("\033[1m" + "\033[93m" + "\nCMS Information Disclosure Vulnerability Tester\n" + "\033[0m")
    target_url = input("\nEnter your target website url: ")
    check_cms_info_disclosure(target_url)

# Main function
if __name__ == "__main__":
    run()
